app/models/transformation_mapping/vm_migration_validator.rb

```diff
@@ -128,6 +133,26 @@ def mapped_lans
     @mapped_lans ||= Lan.where(:id => @mapping.transformation_mapping_items.where(:source_type => 'Lan').select(:source_id))
   end
 
+  # rubocop: disable Naming/UncommunicativeMethodParamName
+  def destination_cluster(vm)
+    @mapping.transformation_mapping_items.find_by(:source_type => 'EmsCluster', :source_id => vm.ems_cluster.id).destination
+  end
+
+  def validate_vm_name(vm)
+    send("validate_vm_name_#{destination_cluster(vm).ext_management_system.emstype}", vm) ? VM_VALID : VM_UNSUPPORTED_NAME
```

Comment:
Not a fan of this, would much rather we added a `VmOrTemplate.validate_name` class method which the providers can implement.  This could then be reused anywhere we needed to validate a name

Reply:
@agrare This would require a bigger change, with a higher risk. We wanted that in Hammer.
If it's a mandatory change for you, I'll remove the `hammer/yes` label.